File: r/euler1_2.r
# Euler1 in R

myMap    <- function(n) n
myFilter <- function(n) ifelse(n%%3 == 0 || n%%5 == 0, n, 0)
myReduce <- function(m,n) m+n

euler1 <- function(n) {
    mapped   = Map(myMap, 1:n)
    filtered = Filter(myFilter, mapped)
    Reduce(myReduce, filtered)
}

euler1(999)

File: f#/euler1_2.fs
(* Euler1 in F# *)

let rec range i j = i :: if i=j then [] else (range (i+j) j);;

let myMap = (List.map (fun x -> x) : (int, int));;

let myFilter = List.filter (fun x -> x % 3 = 0 || x % 5 = 0);;

let myFold = List.fold (+) 0;;

let euler n = range 0 n |> myFilter |> myFold;;

Printf.printf "Euler1 = %d\n" (euler 18);;

